src/OutputPresenter.ts

```diff
@@ -0,0 +1,268 @@
+import * as mobx from "mobx";
+import { templates } from "./Components/Template";
+import type {
+  Mood,
+  TrackEl,
+  LengthEl,
+  DurationTypes,
+} from "./Components/Template";
+
+const MAXLEN = 25;
+
+export class OutputPresenter {
+  constructor() {
+    mobx.makeObservable(this);
+  }
+  @mobx.observable
+  play: boolean = false;
+
+  // same as mediaCounter
+  @mobx.observable
+  playingMedia: number = 0;
+
+  @mobx.observable
+  playedSeconds: number = 0;
+
+  @mobx.observable
+  overallPlayedSeconds: number = 0;
+
+  @mobx.observable
+  currMood: Mood = templates[0];
+  @mobx.observable
+  trackIndex: number = 0;
+  @mobx.observable
+  currTrack: TrackEl = this.currMood.tracks[this.trackIndex];
+
+  @mobx.observable
+  lengthIndex: DurationTypes = "short";
+
+  @mobx.observable
+  currLength: LengthEl = this.currMood.tracks[0][this.lengthIndex];
+
+  @mobx.observable
+  musicLoaded: boolean = false;
+
+  @mobx.observable
+  music: HTMLAudioElement = new Audio(this.currTrack.musicTrack);
+
+  @mobx.computed
+  get isPlaying(): boolean {
+    return this.play;
+  }
+
+  @mobx.computed
+  get currMoodLen(): number {
+    return this.currMood.tracks.length;
+  }
+
+  @mobx.action
+  playVideo(): void {
+    if (this.playedSeconds >= this.currLength.totalDuration) {
+      console.log("resetting video");
+      this.resetVideo();
+    }
+    this.music.play();
+    this.play = true;
+  }
+  @mobx.action
+  pauseVideo(): void {
+    this.music.pause();
+    this.play = false;
+  }
+
+  @mobx.action
+  setPlayingMedia(newMedia: number): void {
+    this.playingMedia = newMedia;
+  }
+
+  @mobx.computed
+  get currPlayingMedia(): number {
+    return this.playingMedia;
+  }
+
+  @mobx.action
+  increPlayedSeconds(seconds: number) {
+    this.playedSeconds += seconds;
+    this.overallPlayedSeconds += seconds;
+    this.adjustSound();
+    if (this.playedSeconds >= this.currLength.slot[this.playingMedia]) {
+      this.playingMedia += 1;
+      this.playedSeconds = 0;
+    }
+  }
+
+  @mobx.action
+  adjustSound() {
+    if (this.music.currentTime >= this.currLength.end - 4) {
+      if (this.music.volume - 0.2 >= 0) {
+        this.music.volume -= 0.2;
+      }
+    } else if (this.music.currentTime - 4 <= this.currLength.start) {
+      if (this.music.volume + 0.1 <= 1) {
+        this.music.volume += 0.1;
+      }
+    }
+  }
+  @mobx.action
+  incrementPlayingMedia(): void {
+    this.playingMedia++;
+  }
+
+  @mobx.computed
+  get currSlotLength(): number {
+    return this.currLength.slot.length;
+  }
+
+  @mobx.computed
+  get totalVideoDuration(): number {
+    return this.currLength.totalDuration;
+  }
+
+  @mobx.computed
+  get currOverallPlayedSeconds(): number {
+    return this.overallPlayedSeconds;
+  }
```

Comment:
this isn't required as `overallPlayedSeconds` is public - you can just reference it directly

Reply:
if this should be removed and access by outputPresenter.overallPlayedSeconds then should other variables like play also be accessed by outputPresenter.play instead of outputPresenter.isPlaying?

Reply:
Yep (although I prefer the `isPlaying` name)
Otherwise you're just renaming the variable, which is an extra step you don't need